source/globalCommands.py

```diff
@@ -3483,8 +3523,74 @@ def script_braille_scrollForward(self, gesture):
 		description=_("Routes the cursor to or activates the object under this braille cell"),
 		category=SCRCAT_BRAILLE
 	)
-	def script_braille_routeTo(self, gesture):
+	def script_braille_routeTo(self, gesture: inputCore.InputGesture) -> None:
+		"""Routes cursor to or activates the object under this braille cell."""
+		# Try to ensure that braille message can be dismissed.
+		if braille.handler.buffer is braille.handler.messageBuffer:
+			braille.handler.routeTo(gesture.routingIndex)
+			return
 		braille.handler.routeTo(gesture.routingIndex)
+		# To proceed braille should be tethered to review cursor which means
+		# that tether is automatic (and review cursor is shown in braille)
+		# or tether is to review cursor.
+		# "Route review cursor and system caret" should be also enabled
+		# in braille settings.
+		if (
+			TetherTo.REVIEW.value != config.conf["braille"]["tetherTo"]

```

Comment:
This puzzles me. This is not going to work when tethering automatically, right?

```suggestion
			self._tether != TetherTo.REVIEW.value
```